src/actions/generalActions.js

```diff
@@ -0,0 +1,57 @@
+const reactToGeneralAction = (model, toolHandler) => {
+	return (action) => {
+		switch (action.type) {
+			case 'MODE_UPDATE':
+				if (model.activityInProgress === true) {
+          toolHandler.cleanTool();
+          model.activityInProgress = false;
+        }
+
```

Comment:
what is the meaning of the variables model.mode and action.mode why should model.mode be set to action.mode followed by updatetool?

What is the difference between action.type and action.mode? 
The code is just too opaque!

Reply:
The way the codebase was set up, the dispatcher takes a model (state store), and an action which is an object. So when a mode is changed, the call to change it through the dispatcher is something like `this.dispatch({ type: 'MODE_UPDATE', mode });`. This is the action dispatched when one of the control buttons is pressed. 

The object passed in that call is referred to as an action. So once dispatched, the controller/actions.js checks if the mode sent by the action is different from the mode held in state. The params in the action object can be renamed to something more clear